### ISSUE TITLE
Popular Items - Duplicate Values

### ISSUE THREAD

**Issue by KendraMBach** [KendraMBach/CIS556_Project#23]
When testing a couple of items, I came across this issue:
![image](https://user-images.githubusercontent.com/25671109/49981661-efb11480-ff26-11e8-962e-d341c13b5cc5.png)

However, it is only present in the popular items category. I have spent a couple of hours trying to reconfigure the HQL but I'm not having any luck due to its limitations. 
So if we think it would be worth our while to fix, I may need some assistance.

**Comment by mferrall** [comment]
I don't think that its worth our while.  Its just too much work for too small of an improvement at this point.

**Comment by KendraMBach** [comment]
Okay, I'm fine with that. I'll focus on the documentation instead.

**Comment by CooperStansbury** [comment]
@KendraMBach not positive, but I think the issue is that the `d.product.id` is included in the query. The `d.product.id` distinguishes products at the size and material level, so it would make sense that two similar items in different sizes would come as separate records in the DAO. This is actually expected behavior in this scenario (even if it is ugly). 

If worth solving, I would want know which products are duplicated from the initial db state. I bet `cat ring` doesn't show up. But if it does, then I have no idea what's going on. 

**NOT A PRIORITY**

```java
   @SuppressWarnings("deprecation")
	public List<Order> listAllOrderItemsForSingleOrder(int orderId) {
        String sql = "Select new " + Order.class.getName() //
                + "(d.id, d.product.id, d.customer.id) "//
                + " from " + Order.class.getName() + " d "//
                + " where d.order.id = :orderId ";
        
        Session session = this.sessionFactory.getCurrentSession();
 
        Query query = session.createQuery(sql);
        
        query.setParameter("orderId", orderId);
 
        return query.list();
    }
    
    
	public List<Product> listAllOrderItemsForAllOrders() {
			
	        String sql = "select d.prodId "//
	                + "from " + Order.class.getName() + " d group by "
	                		+ "d.prodId order by count(d.prodId) desc";
	 
	        Session session = this.sessionFactory.getCurrentSession();
	 
	        Query query = session.createQuery(sql);
	        
	 
	        return (List<Product>) query.list();
    }
    
    
	
 
}
```

**Comment by KendraMBach** [comment]
I must have missed your note earlier. Hmm interesting... considering that, I have something I want to try that may not be related to the code at all. I'll get back to you on the results.
